inc/composer/class-command.php

```diff
@@ -679,7 +679,11 @@ protected function db( InputInterface $input, OutputInterface $output ) {
 
 				break;
 			case 'exec':
-				$query = $input->getArgument( 'options' )[1] ?? null;
+				$options = $input->getArgument( 'options' ) ?? [];
+				array_shift( $options ); // remove the subcommand, we don't need it
+				$query = array_pop( $options ) ?: null; // the query is always the last option
+				$args = count( $options ) > 1 ? implode( ' ', $options ) : ''; // implode all optional options
```

Comment:
Elsewhere in this file the comments are all on their own line before the code they're describing, rather than line-ending comments. I'd suggest maintaining "keep comments on their own line" for consistency with the existing style.